apps/system/js/bluetooth_v2.js

```diff
@@ -0,0 +1,373 @@
+          'bluetooth.enabled': true
+        });
+      }, () => { //reject
+        this.debug('can not get bluetooth adapter');
+      });
+    });
+  },
+
+  /**
+   * update settings value and disable bluetooth hardware
+   *
+   * @private
+   */
+  _disableHandler: function bt__enableHandler() {
+    this.debug('bluetooth settings disabled');
+    this.getAdapter().then((adapter) => {
+      adapter.disable().then(() => { //resolve
+        navigator.mozSettings.createLock().set({
+          'bluetooth.enabled': false
+        });
+      }, () => { //reject
+        this.debug('can not get bluetooth adapter');
+      });
+    });
+  },
+
+  /**
+   * Dispatch events to bluetooth_transfer.js.
+   * while getting bluetooth file transfer start/complete system messages
+   *
+   * @private
+   * @param  {String} msg message
+   * @param  {Object} transferInfo relayed info
+   */
+  _relayMessageEvent: function bt__relayMessageEvent(msg, transferInfo) {
+    this._setProfileConnected(this.Profiles.OPP, true);
+    var evt = document.createEvent('CustomEvent');
+    evt.initCustomEvent(msg,
+      /* canBubble */ true, /* cancelable */ false,
+      {transferInfo: transferInfo});
+    window.dispatchEvent(evt);
+  },
+
+  /**
+   * Maintain connect stat of supported profiles.
+   *
+   * @private
+   * @param  {Object} adapter bluetooth adapter
+   */
+  _updateProfileStat: function bt_updateProfileStat(adapter) {
+    /* for v1, we only support two use cases for bluetooth connection:
+     *   1. connecting with a headset
+     *   2. transfering a file to/from another device
+     * So we need to listen to corresponding events to know we are (aren't)
+     * connected, then summarize to an event and dispatch to StatusBar
+     */
+    // In headset connected case:
+    adapter.addEventListener('hfpstatuschanged', (evt) => {
+      this._setProfileConnected(this.Profiles.HFP, evt.status);
+    });
+
+    adapter.addEventListener('a2dpstatuschanged', (evt) => {
+      this._setProfileConnected(this.Profiles.A2DP, evt.status);
+    });
+
+    adapter.addEventListener('scostatuschanged', (evt) => {
+      this._setProfileConnected(this.Profiles.SCO, evt.status);
+    });
+  },
+
+  /**
+   * BT APIv2: Watch 'onattributechanged' event from
+   * mozBluetooth.defaultAdapter for updating state information.
+   *
+   * @private
+   * @param  {Object} evt event object
+   */
+  _bluetoothAdapterAttrChangeHandler:
+    function bt__bluetoothAdapterAttrChangeHandler(evt) {
+      for (var i in evt.attrs) {
+        switch (evt.attrs[i]) {
+          case 'state':
+            if (this._adapter.state === 'enabled') {
+              this._isEnabled = true;
+              window.dispatchEvent(new CustomEvent('bluetooth-enabled'));
+              // this.icon && this.icon.update();
+            } else if (this._adapter.state === 'disabled') {
+              this._isEnabled = false;
+              window.dispatchEvent(new CustomEvent('bluetooth-disabled'));
```

Comment:
You could update `navigator.mozSettings.createLock().set({'bluetooth.enabled': false});` immediately here.